custom/icds/repeaters/generators/phi.py

```diff
@@ -26,28 +26,45 @@ def _gender(gender):
         return ""
 
     def get_payload(self, repeat_record, case):
-        data = {
-            "beneficaryname": case.name,
-            "fathername": case.get_case_property('fathers_name') or "",
-            "husbandname": case.get_case_property('husbands_name') or "",
-            "mothername": case.get_case_property('mothers_name') or "",
+        data = self._setup_names(case)
+        data.update({
             "gender": self._gender(case.get_case_property('gender')),
             "villagecode": 442639,
             "subdistrictcode": 3318,
             "districtcode": 378,
             "statecode": 22,
-            "namelocal": "",
             "dateofbirth": case.get_case_property('date_of_birth') or "",
             "mobileno": case.get_case_property('mobile_number') or "",
-            "mothernamelocal": "",
-            "fathernamelocal": "",
-            "husbandnamelocal": "",
             "email": "",
             "govt_id_name": "",
             "govt_id_number": ""
-        }
+        })
         return json.dumps(data, cls=DjangoJSONEncoder)
 
+    def _setup_names(self, case):
```

Comment:
@kaapstorm @esoergel i am not proud of this but i don't see another way out.
The API expects to receive english names and local language names in different fields. So that is what this and other relevant methods are trying to achieve.

Reply:
Huh, so if the case name is Ethan, they'd want to receive:
```python
{'beneficaryname': "Ethan", 'name_local': ""}
```
and if the case name is "Éthán"
```python
{'beneficaryname': "", 'name_local': "Éthán"}
```
Is that right?  That's very weird.

Here's another way to structure this if that's what you're looking for:
```python
    def get_payload(self, repeat_record, case):
        data = {}
        data.update(self._setup_names(case))
        ...

    def _setup_names(self, case):
        for key_name, key_name_local, val in [
            ('beneficaryname', 'namelocal', case.name),
            ('mothername', 'mothernamelocal', case.get_case_property('mothers_name')),
            ('fathername', 'fathernamelocal', case.get_case_property('fathers_name')),
            ('husbandname', 'husbandnamelocal', case.get_case_property('husbands_name')),
        ]:
            yield (key_name, val if val.isascii() else '')
            yield (key_name_local, "" if val.isascii() else val)
```

If the thing you don't like about this is the way it obscures what actually makes it in to the payload, you could also structure it like this:
```python
    def get_payload(self, repeat_record, case):
        def _ascii_or_null(val): return val if val.isascii() else ''
        def _unicode_or_null(val): return '' if val.isascii() else val

        return {
            'beneficaryname': _ascii_or_null(case.name),
            'name_local': _unicode_or_null(case.name),
            'mothername': _ascii_or_null(case.get_case_property('mothers_name')),
            'mothernamelocal' _unicode_or_null(case.get_case_property('mothers_name')),
            'fathername': _ascii_or_null(case.get_case_property('fathers_name')),
            'fathernamelocal' _unicode_or_null(case.get_case_property('fathers_name')),
            'husbandname': _ascii_or_null(case.get_case_property('husbands_name')),
            'husbandnamelocal' _unicode_or_null(case.get_case_property('husbands_name')),
            ...
            # other properties
        }
```

Reply:
thanks both.
Tried using the method i added `_has_special_chars` instead of `isascii`. Then noted one thing that the current implementation makes it explicit that only one of the values will be set which is implicit here. So keeping the current implementation also to keep smaller methods.